lib/dry/view/renderer.rb

```diff
@@ -39,21 +39,23 @@ def render(path, scope, &block)
       def chdir(dirname)
         new_paths = paths.map { |path| path.chdir(dirname) }
 
-        self.class.new(new_paths, engine: engine, format: format)
+        self.class.new(new_paths, engines: engines, format: format)
       end
 
       def lookup(name)
-        template_name = "#{name}.#{format}.#{engine}"
+        engines.map do |engine|
+          template_name = "#{name}.#{format}.#{engine}"
 
-        paths.inject(false) { |result, path|
-          result || path.lookup(template_name)
-        }
+          paths.inject(false) { |result, path|
+            result || path.lookup(template_name)
+          }
+        end.reject{|f| f == false}.first
       end
 
       private
 
       def tilt(path)
-        tilts.fetch(path) { tilts[path] = Tilt[engine].new(path, nil, default_encoding: "utf-8") }
+        tilts.fetch(path) { tilts[path] = Tilt.new(path, nil, default_encoding: "utf-8") }
```

Comment:
Letting it infer the template type seems fine to me here. Means we can continue to create only a single tilt instance.